tools/run_tests/filter_pull_request_tests.py

```diff
@@ -0,0 +1,253 @@
+                         'test/distrib/ruby']
+				   
+ends_with_whitelist = ['README.md',
+                       'LICENSE']
+
+# Triggers for core tests
+core_starts_with_triggers = ['test/core']
+
+# Triggers for c++ tests
+cpp_starts_with_triggers = ['src/cpp',
+                            'test/cpp',
+                            'test/distrib/cpp']
+
+# Triggers for c# tests
+csharp_starts_with_triggers = ['src/csharp',
+                               'test/distrib/csharp']
+
+# Triggers for node tests
+node_starts_with_triggers = ['src/node',
+                             'test/distrib/node']
+
+# Triggers for objective-c tests
+objc_starts_with_triggers = ['src/objective-c']
+
+# Triggers for php tests
+php_starts_with_triggers = ['src/php',
+                            'test/distrib/php']
+
+# Triggers for python tests
+python_starts_with_triggers = ['src/python',
+                               'test/distrib/python']
+
+# Triggers for ruby tests
+ruby_starts_with_triggers = ['src/ruby',
+                            'test/distrib/ruby']
+
+
+def _filter_whitelist(whitelist, triggers):
+  """
+  Removes triggers from whitelist
+  :param whitelist: list to remove values from
+  :param triggers: list of values to remove from whitelist
+  :return: filtered whitelist
+  """
+  filtered_whitelist = list(whitelist)
+  for trigger in triggers:
+    if trigger in filtered_whitelist:
+      filtered_whitelist.remove(trigger)
+    else:
+      """
+      If the trigger is not found in the whitelist, then there is likely
+      a mistake in the whitelist or trigger list, which needs to be addressed
+      to not wrongly skip tests
+      """
+      print("ERROR: '%s' trigger not in whitelist. Please fix this!" % trigger)
+  return filtered_whitelist
+
+
+def _get_changed_files(base_branch):
+  """
+  Get list of changed files between current branch and base of target merge branch
+  """
+  # git fetch might need to be called on Jenkins slave
+  # todo(mattkwong): remove or uncomment below after seeing if Jenkins needs this
+  # call(['git', 'fetch'])
+
+  # get file changes between branch and merge-base of specified branch
+  # not combined to be Windows friendly
+  base_commit = check_output(["git", "merge-base", base_branch, "HEAD"]).rstrip()
+  return check_output(["git", "diff", base_commit, "--name-only"]).splitlines()
+
+
+def _can_skip_tests(file_names, starts_with_whitelist=[], ends_with_whitelist=[]):
+  """
+  Determines if tests are skippable based on if all file names do not match
+  any begin or end triggers
+  :param file_names: list of changed files generated by _get_changed_files()
+  :param starts_with_triggers: tuple of strings to match with beginning of file names
+  :param ends_with_triggers: tuple of strings to match with end of file names
+  :return: safe to skip tests
+  """
+  # convert lists to tuple to pass into str.startswith() and str.endswith()
+  starts_with_whitelist = tuple(starts_with_whitelist)
```

Comment:
btw, python has `any` and `all` operators. Using them might result in more readable code.